info/diktat-kotlin-coding-style-guide-en.md

```diff
@@ -709,28 +709,35 @@ All variants of a (private) val logger should be placed in the beginning of the
 
  #### <a name="r3.3"></a>Rule 3.3 Braces must be used in conditional statements and loop blocks
 
-  In `if`, `else`, `for`, `do`, and `while` statements, even if the program body is empty or contains only one statement, braces should be used.
-  In special Kotlin `when` statement no need to use braces for statements with 1 line. Valid example:
+1) In `if`, `else`, `for`, `do`, and `while` statements, even if the program body is empty or contains only one statement, braces should be used.
+In special Kotlin `when` statement no need to use braces for statements with 1 line. Valid example:
+
     ```kotlin
-        when (node.elementType) {
-            FILE -> {
-                checkTopLevelDoc(node)
-                checkSomething()
-             }
-            CLASS -> checkClassElements(node)
-        }
+    when (node.elementType) {
+        FILE -> {
+            checkTopLevelDoc(node)
+            checkSomething()
+         }
+        CLASS -> checkClassElements(node)
+    }
     ```
-  
 **Exception:** *Only* The only exception is ternary operator in Kotlin (it is a single line `if () <> else <>` ) 
 When the entire expression can be 
 
+2) No need to use braces for the body of lambdas and when-conditions (after an arrow)
+  
 Bad example:
 
 ```kotlin
 val value = if (string.isEmpty())  // WRONG!
                 0
             else
                 1
+
```

Comment:
I still think that mixing examples on lambdas and ternary if-else seems misleading for readers.